docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+kind: TemporalConnection
+metadata:
+  name: production-temporal
+  namespace: default
+spec:
+  hostPort: "production.abc123.tmprl.cloud:7233"
+  mutualTLSSecret: temporal-cloud-mtls  # If using mTLS
+---
+apiVersion: temporal.io/v1alpha1
+kind: TemporalConnection
+metadata:
+  name: staging-temporal
+  namespace: default
+spec:
+  hostPort: "staging.abc123.tmprl.cloud:7233"
+  mutualTLSSecret: temporal-cloud-mtls
+```
+
+### Step 3: Prepare Your Worker Code (If Needed)
+
+Most workers will work without changes, but you may need to update your worker initialization code to properly handle versioning:
+
+**Before (Unversioned):**
+```go
+// Worker connects without versioning
+worker := worker.New(client, "my-task-queue", worker.Options{})
+```
+
+**After (Versioned - Optional Enhancement):**
+```go
+// Worker can optionally use build ID from environment
+buildID := os.Getenv("WORKER_BUILD_ID")
+workerOptions := worker.Options{}
+if buildID != "" {
+    workerOptions.BuildID = buildID
+    workerOptions.UseBuildIDForVersioning = true
+}
+worker := worker.New(client, "my-task-queue", workerOptions)
+```
+
+> **Note**: The controller automatically sets `WORKER_BUILD_ID` environment variable, so most workers will work without code changes.
+
+### Step 4: Create Your First TemporalWorkerDeployment
+
+Start with your lowest-risk worker. Convert your existing unversioned Deployment to a `TemporalWorkerDeployment` CRD:
+
+**Current Unversioned Deployment:**
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: payment-processor
+spec:
+  replicas: 3
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: payment-processor:v1.5.2
+        env:
+        - name: TEMPORAL_HOST_PORT
+          value: "production.tmprl.cloud:7233"
+        - name: TEMPORAL_NAMESPACE
+          value: "production"
+```
+
+**New TemporalWorkerDeployment CRD (IMPORTANT: Use Manual strategy initially):**
+```yaml
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: payment-processor
+  labels:
+    app: payment-processor
+spec:
+  replicas: 3
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: production
+  # CRITICAL: Use Manual during initial migration
+  cutover:
+    strategy: Manual
+  sunset:
+    scaledownDelay: 30m
+    deleteDelay: 2h
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: payment-processor:v1.5.2  # Same image as current deployment
+        resources:
+          requests:
+            memory: "512Mi"
+            cpu: "250m"
+        # Note: Controller automatically adds TEMPORAL_* env vars
+```
+
+### Step 5: Deploy the TemporalWorkerDeployment
+
+1. **Create the `TemporalWorkerDeployment` CRD:**
+   ```bash
+   kubectl apply -f payment-processor-versioned.yaml
+   ```
+
+2. **Wait for the version to be registered:**
+   ```bash
+   # Monitor until status shows the version is registered and current
```

Comment:
just a note that if mode is manual, version won't become current. but if mode is progressive (which imo it should be) then it will.